client/pkg/transport/transport.go

```diff
@@ -31,7 +33,17 @@ func NewTransport(info TLSInfo, dialtimeoutd time.Duration) (*http.Transport, er
 	}
 
 	t := &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
+		Proxy: func(req *http.Request) (*url.URL, error) {
```

Comment:
Why not just use the standard `HTTP_PROXY` environment variable? For localhost it you need to just to explicitly set Proxy to `ProxyFromEnvironment`

Reply:
Because of this (comment taken from Golang codebase) 
https://github.com/etcd-io/etcd/pull/17938/files#diff-110c492a19c9cff872b93ff979483976e6c752900453b2b9b83627bdcdfa0026R37

Basically, localhost and 127.0.0.1 will be ignored by Golang when setting those 2 ENV variables (tried and verified that it's indeed not working)

@fuweid and I synced offline and I will put up a fix to further optimize the proxy setup!

Reply:
Nice, sorry for asking without reading the comment, looks like you did your research. Great work!

Reply:
No worries! I should have done a better job explaining all these in the PR description in the first place 🙃

Notes taken!